modules/json/src/smithy4s/http/json/SchemaVisitorJCodec.scala

```diff
@@ -1155,6 +1155,21 @@ private[smithy4s] class SchemaVisitorJCodec(maxArity: Int)
           )
         }
 
+      private[this] val defaults =
+        new util.HashMap[String, Any](
+          documentFields.length << 1,
+          0.5f
+        ) {
+          documentFields.foreach { field =>
+            val decode: Document => Option[Any] =
+              Document.Decoder.fromSchema(field.instance).decode(_).toOption
+            field.getDefault.foreach { d =>
+              val result = decode(d)
+              result.foreach(put(jsonLabel(field), _))
```

Comment:
you have a bug there, as you're using the `jsonLabel` and when looking up you're using the field's label. Moreover, safer to use the field object as a key in the map, as opposed to the label.